src/app/components/driver-list/driver-list.component.ts

```diff
@@ -106,84 +108,117 @@ export class DriverListComponent implements OnInit {
-                                                  <h1>${name}</h1>
-                                                  <h3>Email: ${element.email}</h3>
-                                                  <h3>Phone: ${element.phone}</h3>
-                                              </div>
-                                              <div class="modal-footer">
-                                                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
-                                              </div>
-                                            </div>
-                                         </div>
-                                       </div>
-                                  </div>
-                                  <div class="col-lg-6">
-                                      <div #maps id="gmap" class="img-responsive"></div>
-                                  </div>
-                                </td>
-                                <td class="col">${seats}</td>
-                                </tr>`;
+          if (status !== 'OK') {
+            alert('Error was: ' + status);
+          } 
+          else {
+            
+            // on success, for each driver, do this:
+          
+            /**
+             * 
+             * The architecture of Response is as following:
+             * 
+             * NOTE: 'rows' is an array where each row corresponds to an origin.
+             *        Since we are dealing with SINGLE origin, we can only have 
+             *        value at rows[0]. 
+             * 
+             * {
+             *  originAddress: [],
+             *  destinationAddress: [],
+             *  rows: [
+             *       {
+                *        status: OK,
+                *        duration: {
+                *            value: 123123,
+                *            text: 19 hours 40 mins
+                *        },
+                *        distance: {
+                *            value: 123,
+                *            text: 50 mi
+                *        }
+            *        },
+            * 
+            *        {},
+            *        {},
+            *        etc
+            *    ]
+            * }
+            */
+
+            const originList = response.originAddresses;
+            const destinationList = response.destinationAddresses;
+            // results is Array of objects. Each object has status, duration, and distance.
+            // in our case, there is only 1 object since there is only 1 destination. Voila.
+            const results = response.rows[0].elements;
+            // driver's name (full name)
+            const name = element.name;
+            // driver's car's number of seats
+            const seats = element.seats;
+            // console.log("seats: " + seats);
+
+            this.dataToDisplay.push({
+              "itemLabel": `exampleModalCentered${element.id}`,
```

Comment:
Any reason why you're using quotes for keys here?  We can define these without quotes.

Reply:
Reflex from another language. Removed them as they are indeed unnecessary.